pkg/build/controller/strategy/util.go

```diff
@@ -649,3 +666,59 @@ func setupBlobCache(pod *corev1.Pod) {
 		pod.Spec.InitContainers = initContainers
 	}
 }
+
+// setupBuildVolumes sets up user defined BuildVolumes
+func setupBuildVolumes(pod *corev1.Pod, buildVolumes []buildv1.BuildVolume) error {
+	// if there are no BuildVolumes or the pod is nil,
+	// there is no processing needed, so just return quickly
+	if len(buildVolumes) == 0 || pod == nil {
+		return nil
+	}
+
+	usedUserVolumeMounts := make(map[string]struct{})
+
+	// iterate over existing VolumeMounts and add to the map
+	if len(pod.Spec.Containers) != 0 {
+		for _, vm := range pod.Spec.Containers[0].VolumeMounts {
+			usedUserVolumeMounts[vm.MountPath] = struct{}{}
+		}
+	}
+
+	for _, buildVolume := range buildVolumes {
+		// check for user provided mountPath collisions and return an error if one is found
+		for _, bvm := range buildVolume.Mounts {
+			if _, ok := usedUserVolumeMounts[bvm.DestinationPath]; ok {
+				// fail if a collision is found
+				return fmt.Errorf("user provided BuildVolumeMount path %q collides with VolumeMount path created by the build controller", bvm.DestinationPath)
+			}
+		}
+
+		volumeSource := corev1.VolumeSource{}
+
+		// if no collisions are found go ahead and add the volume/volume mount to the pod/container
+		switch buildVolume.Source.Type {
+		case buildv1.BuildVolumeSourceTypeSecret:
+			volumeSource.Secret = buildVolume.Source.Secret
+			mountSecretVolume(pod, &pod.Spec.Containers[0], strings.ToLower(buildVolume.Source.Secret.SecretName), PathForBuildVolume(buildVolume.Source.Secret.SecretName), buildVolumeSuffix, &volumeSource)
+		case buildv1.BuildVolumeSourceTypeConfigMap:
+			volumeSource.ConfigMap = buildVolume.Source.ConfigMap
+			mountConfigMapVolume(pod, &pod.Spec.Containers[0], strings.ToLower(buildVolume.Source.ConfigMap.Name), PathForBuildVolume(buildVolume.Source.ConfigMap.Name), buildVolumeSuffix, &volumeSource)
+		default:
+			return fmt.Errorf("encountered unsupported build volume source type %q", buildVolume.Source.Type)
+		}
+	}
+
+	return nil
+}
+
+// NameForBuildVolume returns a valid pod volume name for the provided build volume name.
+func NameForBuildVolume(objName string) string {
```

Comment:
yeah let's see how the possible use of https://github.com/openshift/runtime-utils lands in your builder PR for BUILD-87 wrt this method and `PathForBuildVolume`

Reply:
@adambkaplan said "nay" to adding the comments in that repo ... not sure I entirely agree but I'll defer ... I don't seem to have the resolve button for this one on comments I made @coreydaley but if you do feel free to mark this resolved